test-suite/src/test/scala/org/scalajs/testsuite/junit/ScalaJSJUnitAssertionsTest.scala

```diff
@@ -0,0 +1,304 @@
+  }
+
+  @Test
+  def testAssertSame(): Unit = {
+    // Setup
+    val obj = new Object()
+    val str = "abcd"
+    val nullRef: AnyRef = null
+
+    def testAssertion(expected: AnyRef, actual: AnyRef, equals: Boolean = true): Unit = {
+      testIfAsserts(assertSame("References where not equal", expected, actual), equals)
+      testIfAsserts(assertSame(expected, actual), equals)
+      testIfAsserts(assertNotSame("References where equal", expected, actual), !equals)
+      testIfAsserts(assertNotSame(expected, actual), !equals)
+    }
+
+    // Tests
+    testAssertion(obj, obj)
+    testAssertion(str, str)
+    testAssertion(nullRef, nullRef)
+
+    testAssertion(new Object, new Object, notEquals)
+  }
+
+  @Test
+  def testAssertEquals(): Unit = {
+
+    // Setup
+    val obj = new Object()
+    val str = "abcd"
+    val nullRef: AnyRef = null
+
+    // Object equality tests
+    def testAssertion(expected: AnyRef, actual: AnyRef, equals: Boolean = true): Unit = {
+      testIfAsserts(assertEquals(s"Asserting $expected == $actual", expected, actual), equals)
+      testIfAsserts(assertEquals(expected, actual), equals)
+      testIfAsserts(assertNotEquals(s"Asserting $expected != $actual", expected, actual), !equals)
+      testIfAsserts(assertNotEquals(expected, actual), !equals)
+    }
+
+    testAssertion(nullRef, nullRef)
+    testAssertion(obj, obj)
+    testAssertion(str, str)
+    testAssertion(new Object, null, notEquals)
+    testAssertion(null, new Object, notEquals)
+    testAssertion(new Object, new Object, notEquals)
+
+    testAssertion("", "")
+    testAssertion("42", "42")
+    testAssertion("asdfasfsafs", "asdfasfsafs")
+    testAssertion(List(1, 2, 3), List(1, 2, 3))
+    testAssertion(List(1L, 2L, 3L), List(1L, 2L, 3L))
+    testAssertion(Vector(1L, 2L, 3L), List(1L, 2L, 3L))
+    testAssertion((1, 2, 3), (1, 2, 3))
+    testAssertion("", "d", equals = false)
+    testAssertion("42", "1", equals = false)
+    testAssertion("asdfasfsafs", "asdfuhafs", notEquals)
+    testAssertion(List(1, 2, 3), List(1, 2, 4), notEquals)
+    testAssertion(List(1L, 2L, 3L), List(1L, 2L, 4L), notEquals)
+    testAssertion(Vector(1L, 2L, 3L), List(1L, 2L, 4L), notEquals)
+    testAssertion((1, 2, 3), (1, 2, 4), notEquals)
+
+    // Byte equality tests
+    def testByteAssertion(expected: Byte, actual: Byte, equals: Boolean = true): Unit = {
+      testIfAsserts(assertEquals(s"Asserting $expected == $actual", expected, actual), equals)
+      testIfAsserts(assertEquals(expected, actual), equals)
+      testIfAsserts(assertNotEquals(s"Asserting $expected != $actual", expected, actual), !equals)
+      testIfAsserts(assertNotEquals(expected, actual), !equals)
+    }
+    testByteAssertion(0, 0)
+    testByteAssertion(42, 42)
+    testByteAssertion(-42, -42)
+    testByteAssertion(Byte.MinValue, Byte.MinValue)
+    testByteAssertion(Byte.MaxValue, Byte.MaxValue)
+    testByteAssertion(1, 2, notEquals)
+
+
+    // Char equality tests
```

Comment:
I suggest putting each of these headings (for each primitive data type) in their own `@Test` methods instead.